ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -26,13 +32,60 @@ function identity<T>(value: T) {
     return () => value;
 }
 
+// USED FOR MANUAL TESTING
+// const timeSpammer:Observable<string> = new Observable((subscriber) => {
+//   setInterval(() => {
+//     subscriber.next('time="2022-11-27T04:07:37.291Z" level=info msg="running spammer" argo=true\n');
+//   }, 2000);
+// });
+
+interface ParsedTime {
+    quoted: string;
+    fullstring: string;
+}
+// extract the time field from a string
+const parseTime = (formattedString: string): undefined | ParsedTime => {
+    const re = new RegExp('time="(.*?)"');
+    const table = re.exec(formattedString);
+    if (table === null || table.length !== 2) {
+        return undefined;
+    }
+    return {quoted: table[1], fullstring: table[0]};
+};
+
+const parseAndTransform = (formattedString: string, timezone: string) => {
+    const maybeTime = parseTime(formattedString);
+    if (maybeTime === undefined) {
+        return formattedString;
+    }
+
+    try {
+        const newTime = moment.tz(maybeTime.quoted, timezone).format('YYYY-MM-DDTHH:mm:ss z');
+        const newFormattedTime = `time=\"${newTime}\"`;
+        const newFormattedString = formattedString.replace(maybeTime.fullstring, newFormattedTime);
+        return newFormattedString;
+    } catch {
+        return formattedString;
+    }
+};
+
 export const WorkflowLogsViewer = ({workflow, nodeId, initialPodName, container, archived}: WorkflowLogsViewerProps) => {
     const [podName, setPodName] = useState(initialPodName || '');
     const [selectedContainer, setContainer] = useState(container);
     const [grep, setGrep] = useState('');
     const [error, setError] = useState<Error>();
     const [loaded, setLoaded] = useState(false);
     const [logsObservable, setLogsObservable] = useState<Observable<string>>();
+    // timezone used for ui rendering only
+    const [uiTimezone, setUITimezone] = useState<string>(DEFAULT_TZ);
+    // timezone used for timezone formatting
+    const [timezone, setTimezone] = useLocalStorage<string>(TZ_LOCALSTORAGE_KEY, DEFAULT_TZ);
+    // list of timezones moment tz supports
+    const [timezones, setTimezones] = useState<string[]>([]);
```

Comment:
is there a typo on line 83? can you clarify this comment?

Reply:
No, I just meant that it stores the list of timezones the library supports. I abbreviated timezone with tz. 
I'll make it clearer now.